apm-agent-plugins/apm-apache-httpclient/apm-apache-httpclient3-plugin/src/main/java/co/elastic/apm/agent/httpclient/v3/HttpClient3Instrumentation.java

```diff
@@ -82,54 +80,46 @@ public static class HttpClient3Advice {
         @Advice.OnMethodEnter(suppress = Throwable.class, inline = false)
         public static Object onEnter(@Advice.Argument(0) HttpMethod httpMethod,
                                      @Advice.FieldValue(value = "hostConfiguration") HostConfiguration hostConfiguration) {
-            final AbstractSpan<?> parent = TracerAwareInstrumentation.tracer.getActive();
-            if (parent == null) {
-                return null;
-            }
-
-            String host;
-            String uri;
-            String protocol;
-            int port;
-            URI httpClientURI = null;
-            try {
-                httpClientURI = httpMethod.getURI();
-                host = hostConfiguration.getHost();
-                port = hostConfiguration.getPort();
-                protocol = hostConfiguration.getProtocol().getScheme();
-                uri = httpClientURI.toString();
-            } catch (Exception e) {
+            Span<?> span = null;
+            final ElasticContext<?> parent = TracerAwareInstrumentation.tracer.currentContext();
+            if (parent.getSpan() != null) {
```

Comment:
[nitpicking] keeping the shortcut return when there is no span avoids one level of if nesting.

Reply:
Sorry, should have mentioned it in the PR description (updated it now):
This PR also changes that context propagation always happens, even if there is no active span.
E.g. if someone just updated the baggage, we want to still create baggage headers, even if no trace-context will be propagated. Baggage is conceptually independent from tracing.

For this reason I had to remove the early exits in order to make sure that the context propagation part is always reached.